### PR TITLE
`ci`: Add shellcheck step to lint job

### PR DESCRIPTION
**Pull request by marun**
# #lintallthethings!

We lint code, why not shell scripts? In addition to simplifying the avoidance of common shell errors, the [shellcheck wiki](https://www.shellcheck.net/wiki) has lots of documentation for all the error types it checks for.

### Testing

In addition to CI, local testing on arm64 linux and macos verified the logic for downloading and using a local (vs one in the path) shellcheck.

Related: [AvalancheGo PR](https://github.com/ava-labs/avalanchego/pull/2650)

### CODE REVIEW

**Review by joshua-kim — DISMISSED** [review]

File: cmd/evm/transition-test.sh
Comment (ceyonur) [inline cmd/evm/transition-test.sh:1]: this script is fetched from upstream repo. I wonder if we should exclude the spellcheck for any upstream script.

File: metrics/validate.sh
Comment (ceyonur) [inline metrics/validate.sh:1]: same comment about upstream scripts.

**Review by ceyonur — COMMENTED** [review]
some scripts are from upstream repo, maybe we should exclude them from spellcheck. other changes LGTM.

**Review by ceyonur — COMMENTED** [review]
some scripts are from upstream repo, maybe we should exclude them from spellcheck. other changes LGTM.

**Review by darioush — APPROVED** [review]

**Review by ceyonur — APPROVED** [review]

**Pull request merged**